IntegrationTests/tests_01_allocation_counters/test_01_resources/test_stream_teardown_100_concurrent.swift

```diff
@@ -158,29 +156,68 @@ fileprivate class SendGoawayHandler: ChannelInboundHandler {
     )
 
     private let expectedStreams: Int
-    private var seenStreams: Int
+    private let seenStreams = ManagedAtomic<Int>(0)
 
     init(expectedStreams: Int) {
         self.expectedStreams = expectedStreams
-        self.seenStreams = 0
     }
 
     func userInboundEventTriggered(context: ChannelHandlerContext, event: Any) {
         if event is NIOHTTP2StreamCreatedEvent {
-            self.seenStreams += 1
-            if self.seenStreams == self.expectedStreams {
+            self.seenStreams.wrappingIncrement(ordering: .sequentiallyConsistent)
+            if seenStreams.load(ordering: .sequentiallyConsistent) == self.expectedStreams {
```

Comment:
You can combine the two atomic operations here by using `wrappingIncrementThenLoad`